src/PrefixTree.java

```diff
@@ -15,28 +15,27 @@ private static class TraverseResult {
     }
 
     public boolean contains(CharSequence word) {
-        TraverseResult helper = traverse(word);
-        if (helper.index == -1) {
+        TraverseResult traverseResult = traverse(word);
+        if (traverseResult.index == word.length() && traverseResult.node.isFinal()) {
             System.out.println("Tree contains this word - " + word);
             return true;
-        } else {
-            System.out.println("Tree doesn't contain this word - " + word);
-            return false;
         }
+        System.out.println("Tree doesn't contain this word - " + word);
+        return false;
     }
 
     public boolean add(CharSequence word) {
-        TraverseResult helper = traverse(word);
-        Node currentNode = helper.node;
-        if (helper.index == -1) {
+        TraverseResult traverseResult = traverse(word);
+        Node currentNode = traverseResult.node;
+        if (traverseResult.index == word.length() && traverseResult.node.isFinal()) {
             System.out.println("Word has been added already. " + word);
             return false;
         }
         if (word.length() == 1) {
             currentNode.setFinal(true);
             return true;
         }
```

Comment:
I don't see a reason to treat one char words any different. If this is some kind of optimization attempt, keep in mind the words of Donald Knuth: "premature optimization is the root of all evil".